src/login.js

```diff
@@ -122,14 +127,37 @@ async function askAndFillVerificationCodeIfFieldExists(page) {
         // used for 2FA this path is skipped.
         if (verificationCode.verificationCode.length > 5) {
             await page.type(verificationCodeSelector, verificationCode.verificationCode);
-            await Promise.all([page.click('a.button-action'), page.waitForNavigation()]);
-            // TODO: handle invalid verification code entered response
+
+            try {
+                // The 2FA-page only navigates if the code is valid.
+                await Promise.all([page.click('a.button-action'), page.waitForNavigation({ timeout: 5000 })]);
+            }
+            catch(error) {
+                let validationErrors = await getValidationErrors(page);
+                if (validationErrors.length > 0) {
+                    console.warn(`2FA validation error(s): ${validationErrors.join("\n")}`);
+                    await askAndFillVerificationCodeIfFieldExists(page);
```

Comment:
Seems like this causes an infinite loop if the 2FA code is invalid every time. I would prefer a maximum number of try's.